coderd/workspaceagents_test.go

```diff
@@ -74,6 +74,53 @@ func TestWorkspaceAgent(t *testing.T) {
 		_, err = client.WorkspaceAgent(ctx, workspace.LatestBuild.Resources[0].Agents[0].ID)
 		require.NoError(t, err)
 	})
+	t.Run("Timeout", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, &coderdtest.Options{
+			IncludeProvisionerDaemon: true,
+		})
+		user := coderdtest.CreateFirstUser(t, client)
+		authToken := uuid.NewString()
+		tmpDir := t.TempDir()
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+			Parse:           echo.ParseComplete,
+			ProvisionDryRun: echo.ProvisionComplete,
+			Provision: []*proto.Provision_Response{{
+				Type: &proto.Provision_Response_Complete{
+					Complete: &proto.Provision_Complete{
+						Resources: []*proto.Resource{{
+							Name: "example",
+							Type: "aws_instance",
+							Agents: []*proto.Agent{{
+								Id:        uuid.NewString(),
+								Directory: tmpDir,
+								Auth: &proto.Agent_Token{
+									Token: authToken,
+								},
+								ConnectionTimeout:  1,
```

Comment:
nit: question about the naming conversion, it seems that `ConnectionTimeout` is reserved only for seconds, should we name this `ConnectionTimeoutSec`?  I hope that we won't need a 500ms timeout in the future :)

Reply:
Not a bad suggestion at all. I'm a bit vary of doing it though since this value is propagated through `terraform-provider-coder`, I was following the previous convention for how an int/second is represented in the code (e.g. `codersdk.Healthcheck.Interval`).

We do have other places in the code that utilizes `*Millis`/`_ms`, however. So there is precedent for both.

Reply:
> So there is precedent for both

Good for me, thanks!

Reply:
It might be worth opening an issue to standardize these fields @mafredri... it makes me significantly more nervous about working with code with unspecified durations. The change of course doesn't need to happen in this PR though!

Reply:
I'll create the issue for standardization tomorrow, I did however push the change for this PR in [`a182cdf`](https://github.com/coder/coder/pull/4937/commits/a182cdf603a18c862d4af5898bbcbd5dbe0ff686) already, turned out much better. Thanks for pushing for this change both of you. 👍🏻